python/applications/ranking/data/scripts/create_dataset.py

```diff
@@ -0,0 +1,148 @@
+import os
+import random
+import traceback
+import datetime
+import pandas as pd
+import numpy as np
+from logging import Logger
+
+from ml4ir.io import file_io
+from ml4ir.io import logging_utils
+from ml4ir.features.feature_config import parse_config, FeatureConfig
+
+
+def create_dataset(data_dir: str,
+                   data_out_dir: str,
+                   feature_config_src: str,
+                   max_num_records: int = 5,
+                   num_samples: int = 10,
+                   random_state: int = 123,
+                   logger: Logger = None):
+    """
+    1. Loads example data
+    2. Builds specified synthetic data size by sampling from example data
+    3. Adds catastrophic failures specifically
+    4. For now, write out to CSV. In future could return df directly
+    """
+
+    # Setup logging
+    logs_dir: str = os.path.join('logs', 'test_run_id')
+    file_io.make_directory(logs_dir, clear_dir=True, log=None)
+    logger: Logger = setup_logging()
+    logger.info('Logging initialized. Saving logs to : {}'.format(logs_dir))
+    logger.info('Run ID: {}'.format('test_run_id'))
+
+    try:
+        # Set seeds
+        set_seeds(random_state)
+        logger.info('Set seeds with initial random state {}'.format(random_state))
+
+        # Load and parse feature config #TODO check on tfrecord_type
+        feature_config: FeatureConfig = parse_config(tfrecord_type='', feature_config=feature_config_src, logger=logger)
+        logger.info("Feature config parsed and loaded")
+
+        # Create output location
+        if not os.path.exists(data_out_dir):
+            os.makedirs(data_out_dir)
```

Comment:
@calliebradley You should be using `ml4ir.io.file_io` for all this. If there's no method already to create, I would encourage you to add it. I probably should do a pass through to look at where I've used python methods directly.

Reply:
Thanks @lastmansleeping, good point and changed to use `file_io.make_directory`. Question about that method though - why does it take `dir_path` as an argument and also return `dir_path` ? Seems like the return object is unnecessary?